### ISSUE TITLE
snowflake_view crashes on import of existing view

### ISSUE THREAD

**Issue by lastlee** [Snowflake-Labs/terraform-provider-snowflake#2640]
### Terraform CLI and Provider Versions

Terraform 1.5.7
Snowflake 0.87.2

### Terraform Configuration

```terraform
resource "snowflake_view" "secure_view" {
  database = snowflake_database.subscription.name
  schema   = snowflake_schema.subscription_classification.name
  name     = var.classification_table
  comment  = "Classification View of the union of classification tables"

  statement  = <<-SQL
            select * from AB1_SUBSCRIPTION.CLASSIFICATION.CLASSIFICATION
              union
            select * from AB2_SUBSCRIPTION.CLASSIFICATION.CLASSIFICATION
SQL
  or_replace = true
  is_secure  = true
}
```

### Expected Behavior

The import of `'SUBSCRIPTION|CLASSIFICATION|CLASSIFICATION'` should import successfully

### Actual Behavior

```
snowflake_view.secure_view: Importing from ID "SUBSCRIPTION|CLASSIFICATION|CLASSIFICATION"...
snowflake_view.secure_view: Import prepared!
  Prepared snowflake_view for import
snowflake_view.secure_view: Refreshing state... [id=SUBSCRIPTION|CLASSIFICATION|CLASSIFICATION]
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more
│ details.
╵


Stack trace from the terraform-provider-snowflake_v0.87.2 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 83 [running]:
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake.(*ViewSelectStatementExtractor).consumeToken(0xc000edf068, {0x208e28a, 0x6})
        github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake/parser.go:129 +0x14f
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake.(*ViewSelectStatementExtractor).Extract(0xc000edf068)
        github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake/parser.go:30 +0x92
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources.ReadView(0xc0005fed00, {0x1cc21a0?, 0xc000288a20?})
        github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/resources/view.go:180 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x243d388?, {0x243d388?, 0xc001415200?}, 0xd?, {0x1cc21a0?, 0xc000288a20?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:783 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000a3fce0, {0x243d388, 0xc001415200}, 0xc000ce8750, {0x1cc21a0, 0xc000288a20})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:1089 +0x552
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc000694120, {0x243d388?, 0xc0014150b0?}, 0xc000854b40)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:663 +0x48a
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ReadResource({{0x24531a0?, 0xc000694120?}}, {0x243d388?, 0xc0014150b0?}, 0xc000854800?)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:176 +0x225
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ReadResource(0x243d3c0?, {0x243d388?, 0xc001414db0?}, 0xc000854800)
        github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ReadResource.go:35 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc0002dedc0, {0x243d388?, 0xc0014143c0?}, 0xc00111e120)
        github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/tf6server/server.go:787 +0x48b
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x2014140?, 0xc0002dedc0}, {0x243d388, 0xc0014143c0}, 0xc0005fe000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:467 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001c2d20, {0x243d388, 0xc0014141b0}, {0x244e480, 0xc0006829c0}, 0xc001420120, 0xc000ae46f0, 0x343ed88, 0x0)
        google.golang.org/grpc@v1.60.0/server.go:1372 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0001c2d20, {0x244e480, 0xc0006829c0}, 0xc001420120)
        google.golang.org/grpc@v1.60.0/server.go:1783 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.60.0/server.go:1016 +0x59
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 30
        google.golang.org/grpc@v1.60.0/server.go:1027 +0x115

Error: The terraform-provider-snowflake_v0.87.2 plugin crashed!
```

### Steps to Reproduce

1. `terraform import snowflake_view.secure_view 'SUBSCRIPTION|CLASSIFICATION|CLASSIFICATION'`

### How much impact is this issue causing?

High

**Comment by sfc-gh-asawicki** [comment]
Hey @lastlee. Thanks for reaching out to us.

Can you share the output of the SQL SHOW command for the view you are trying to import?

**Comment by lastlee** [comment]
```
created_on	name	reserved	database_name	schema_name	owner	comment	text	is_secure	is_materialized	owner_role_type	change_tracking
2023-04-17 14:19:56.638 -0700	CLASSIFICATION		SUBSCRIPTION	CLASSIFICATION	APP_SNOWFLAKE_REPORTING_TREX_ADMINS	Classification View of the union of classification tables	CREATE OR REPLACE SECURE VIEW "CLASSIFICATION" comment = 'Classification View of the union of classification tables' AS select * from AB1_SUBSCRIPTION.CLASSIFICATION.CLASSIFICATION    union   select * from AB2_SUBSCRIPTION.CLASSIFICATION.CLASSIFICATION        	true	false	ROLE	OFF
```

**Comment by sfc-gh-asawicki** [comment]
Thanks! I will try reproducing it and get back to you (probably next week).

**Comment by sfc-gh-asawicki** [comment]
Okay, I was able to find the source of the problem.

As https://docs.snowflake.com/en/sql-reference/sql/create-view#usage-notes state:
> By design, the [SHOW VIEWS](https://docs.snowflake.com/en/sql-reference/sql/show-views) command does not provide information about secure views. To view information about a secure view, you must use the [VIEWS](https://docs.snowflake.com/en/sql-reference/info-schema/views) view in the Information Schema and you must use the role that owns the view.

Your view is secure. Therefore, to import it you have to use the view owner role.

I don't think this is a bug in the provider, this is a Snowflake-documented behavior. Either way, provider execution should not end in panic. So:
- We will add the proper description to the snowflake_view resource documentation.
- We will handle the empty `text` gracefully.
- We won't add any workaround custom logic for importing/reading secure views.

**Comment by lastlee** [comment]
Ah, that makes sense. I was just digging into some other issues and noticed the role our provider has defined changed. Good catch, and yeah -- more insightful handling of these cases (and probably for other resources being imported under a different owner (?)) would be good to have handled.

Thanks!

**Comment by sfc-gh-asawicki** [comment]
Hey @lastlee. Changes released in v0.88.0.

Please check and close the issue if the problem is solved for you.

**Comment by lastlee** [comment]
Thanks @sfc-gh-asawicki. I'm not able to test this anymore on our current Snowflake accounts as (thanks to your investigation) we updated the ownership on the view to ACCOUNTADMIN. It does look like your changes should fix this issue for us, should we have the ownership misalignment with the provider.
